R/engine.R

```diff
@@ -174,6 +174,55 @@ get_engine_opts = function(opts, engine, fallback = '') {
 
 get_engine_path = function(path, engine) get_engine_opts(path, engine, engine)
 
+# engine.opts = list(command, file, args, args1, args2)
+eng_exec = function(options) {
+  opts = options$engine.opts %n% options[c('command', 'file', 'ext', 'args', 'args1', 'args2')]
+  if (!is.character(cmd <- opts$command)) stop(
+    "The command of the 'exec' engine was not specified."
+  )
+
+  # default options
+  opts2 = list(
+    ext = identity, file = function(code, file) {
+      write_utf8(options$code, file)
+      file
+    }, args = function(code, file) {
+      file
+    }, clean = function(file) {
+      unlink(file)
+    }
+  )
+
+  opts = merge_list(opts2, opts)
+  cmd2 = basename(cmd)  # in case command is a full path
+  ext = opts$ext(cmd2)  # file extension
+  f = if (is.function(opts$file)) {
+    f = wd_tempfile(cmd2, paste0('.', ext))
+    if (is.function(opts$clean)) on.exit(opts$clean(f), add = TRUE)
+    opts$file(options$code, f)
+  }
+  a = c(opts$args1, if (is.function(opts$args)) opts$args(options$code, f), opts$args2)
+
+  out = if (options$eval) {
+    if (options$message) message('running: ', paste(c(cmd, a), collapse = ' '))
+    f2 = wd_tempfile(cmd2)  # capture stderr
```

Comment:
It seems like `f2` will be created in all case. I have those tempfile left after running the example. 
This happens at least on Windows. `file.size(f2)` is 0 though, so it creates no error. 

I think we need to unlink on exit

```suggestion
    f2 = wd_tempfile(cmd2)  # capture stderr
    on.exit(unlink(f2), add = TRUE)
```